### PR TITLE
[1.15.x] Added entity nameplate rendering event hook

### PR DESCRIPTION
**Pull request by LeScooter**
Adds a new event hook RenderNameplateEvent, which lets modders change, cancel or force the rendering of the nameplate and its contents.

### CODE REVIEW

File: patches/minecraft/net/minecraft/client/renderer/entity/EntityRenderer.java.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
+--- a/net/minecraft/client/renderer/entity/EntityRenderer.java
++++ b/net/minecraft/client/renderer/entity/EntityRenderer.java
+@@ -15,6 +15,7 @@
+ import net.minecraft.world.LightType;
+ import net.minecraftforge.api.distmarker.Dist;
+ import net.minecraftforge.api.distmarker.OnlyIn;
++import net.minecraftforge.eventbus.api.Event;
```
Comment (diesieben07) [inline patches/minecraft/net/minecraft/client/renderer/entity/EntityRenderer.java.patch:7]: No imports in patch files.

File: patches/minecraft/net/minecraft/client/renderer/entity/EntityRenderer.java.patch [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ @OnlyIn(Dist.CLIENT)
+ public abstract class EntityRenderer<T extends Entity> {
+@@ -54,8 +55,10 @@
+    }
+ 
+    public void func_225623_a_(T p_225623_1_, float p_225623_2_, float p_225623_3_, MatrixStack p_225623_4_, IRenderTypeBuffer p_225623_5_, int p_225623_6_) {
+-      if (this.func_177070_b(p_225623_1_)) {
+-         this.func_225629_a_(p_225623_1_, p_225623_1_.func_145748_c_().func_150254_d(), p_225623_4_, p_225623_5_, p_225623_6_);
++      net.minecraftforge.client.event.RenderNameplateEvent renderNameplateEvent = new net.minecraftforge.client.event.RenderNameplateEvent(p_225623_1_,p_225623_1_.func_145748_c_().func_150254_d());
++      if(net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(renderNameplateEvent)) return;
```
Comment (diesieben07) [inline patches/minecraft/net/minecraft/client/renderer/entity/EntityRenderer.java.patch:18]: Formatting (missing whitespace)

File: patches/minecraft/net/minecraft/client/renderer/entity/EntityRenderer.java.patch [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
++      if (!renderNameplateEvent.getResult().equals(Event.Result.DENY) && (renderNameplateEvent.getResult().equals(Event.Result.ALLOW) || this.func_177070_b(p_225623_1_))) {
```
Comment (diesieben07) [inline patches/minecraft/net/minecraft/client/renderer/entity/EntityRenderer.java.patch:19]: Why is this event both cancelable _and_ has a result? `Result.DENY` already covers the "cancel it" case. Your event class is also missing `@HasResult` if you want a result.
Why `equals` for enums?

**Review by diesieben07 — CHANGES_REQUESTED** [review]
Do we really want yet another event that fires _every frame_ for _every entity_?

**Comment by LeScooter** [comment]
Creating a new mended PR

**Comment by diesieben07** [comment]
> Creating a new mended PR

NO! Just update your local branch. You can even change the source branch for this PR.

**Pull request closed without merging**